### PR TITLE
Add common methods from bits.dart to binary.dart

### PR DESCRIPTION
**Pull request by kharland**

### CODE REVIEW

File: binary/lib/binary.dart [hunk 1]
```diff
@@ -233,6 +233,29 @@ class Integral implements Comparable<Integral> {
     return result;
   }
 
+  /// Returns true iff [number] is negative, else false.
+  bool isNegative(int number) =>
+      isSigned ? getBit(number, length - 1) == 1 : false;
+
+  /// Returns [1] if the [result] of an addition produced a carry bit, else 0.
```
Comment (matanlurey) [inline binary/lib/binary.dart]: Update this and comments below, it returns true/false now, not 1 or 0.

Comment (kharland) [inline binary/lib/binary.dart]: done

File: binary/lib/binary.dart [hunk 2]
```diff
@@ -233,6 +233,29 @@ class Integral implements Comparable<Integral> {
     return result;
   }
```
Comment (matanlurey) [inline binary/lib/binary.dart]: Sorry one more thing - to be consistent should we have top-level unchecked methods too?

Obviously this doesn't make sense for `hasCarryBit`.

File: binary/lib/binary.dart [hunk 3]
```diff
@@ -233,6 +233,29 @@ class Integral implements Comparable<Integral> {
...
+      isSigned ? getBit(number, length - 1) == 1 : false;
+
+  /// Returns true if the [result] of an addition produced a carry bit.
+  ///
+  /// A carry bit is produced during a two's compliment addition if the unmasked
+  /// result is greater than [Integral.max].
+  bool hasCarryBit(int result) => result > max;
+
+  /// Returns true if [op1] + [op2] produced a signed overflow in [result].
+  bool isAddOverflow(int op1, int op2, int result) =>
```
Comment (matanlurey) [inline binary/lib/binary.dart]: `doesAddOverflow`?

File: binary/lib/binary.dart [hunk 4]
```diff
@@ -233,6 +233,29 @@ class Integral implements Comparable<Integral> {
...
+      isNegative(op1) == isNegative(op2) &&
+      isNegative(result) != isNegative(op1);
+
+  /// Returns true if [op1] - [op2] produced a signed overflow in [result].
+  bool isSubOverflow(int op1, int op2, int result) =>
```
Comment (matanlurey) [inline binary/lib/binary.dart]: `doesSubOverflow`?

File: binary/lib/binary.dart (hunk 3 above)
Comment (kharland) [inline binary/lib/binary.dart]: Sure. will update.

File: binary/lib/binary.dart (hunk 4 above)
Comment (kharland) [inline binary/lib/binary.dart]: same.

File: binary/lib/binary.dart (hunk 2 above)
Comment (kharland) [inline binary/lib/binary.dart]: doesAddOverflow and doesSubOverflow call to isNegative which requires some Integral.max. If we made these top-level they'd need an extra argument, the Integral.  That doesn't match the pattern of the rest of the methods but If you think it's needed, I don't see why not.

Comment (kharland) [inline binary/lib/binary.dart]: Can always add later. Will merge this so that I can work on #35

**Pull request merged**